### PR TITLE
chore: add log before executing external command

### LINKED ISSUE #3275: Add log output before executing external command in GetCredentials
When Terragrunt executes an external command in GetCredentials it would be helpful to log the instance before it is attempted.

### PR DESCRIPTION
**Pull request by Resonance1584**
## Description

Fixes #3275.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added log before executing external auth-provider-cmd

### CODE REVIEW

File: cli/commands/terraform/creds/providers/externalcmd/provider.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func (provider *Provider) GetCredentials(ctx context.Context) (*providers.Creden
 		args = parts[1:]
 	}
 
+	provider.terragruntOptions.Logger.Infof("Executing %s to obtain Terragrunt credentials", provider.Name())
```
Comment (Resonance1584) [inline cli/commands/terraform/creds/providers/externalcmd/provider.go:46]: Open to any suggestions for wording / format - my goal here is to help explain why a particular command started to execute when viewing log output with many steps.

**Review by levkohimins — APPROVED** [review]
LGTM!

File: cli/commands/terraform/creds/providers/externalcmd/provider.go (hunk 1 above)
Comment (levkohimins) [reply cli/commands/terraform/creds/providers/externalcmd/provider.go:46]: Everything is great, thanks for your contribution!

**Review by levkohimins — APPROVED** [review]
LGTM!

**Review by yhakbar — APPROVED** [review]

**Pull request merged**